src/mailbox.cpp

```diff
@@ -11,64 +11,78 @@
 
 extern Game g_game;
 
-ReturnValue Mailbox::queryAdd(int32_t, const Thing& thing, uint32_t, uint32_t, Creature*) const
+static bool canSend(std::shared_ptr<const Item> item)
 {
-	const Item* item = thing.getItem();
-	if (item && Mailbox::canSend(item)) {
+	return item->getID() == ITEM_PARCEL || item->getID() == ITEM_LETTER;
+}
+
+ReturnValue Mailbox::queryAdd(int32_t, std::shared_ptr<const Thing> thing, uint32_t, uint32_t,
+                              std::shared_ptr<Creature>) const
+{
+	auto item = thing->getItem();
```

Comment:
Could this be `const auto`
```suggestion
	const auto item = thing->getItem();
```